x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/TokenService.java

```diff
@@ -278,43 +281,19 @@ private void createUserToken(String userTokenId, Authentication authentication,
-                final String documentId = getTokenDocumentId(userToken);
-                IndexRequest request =
-                        client.prepareIndex(SECURITY_INDEX_NAME, SINGLE_MAPPING_NAME, documentId)
+                request = client.prepareIndex(SECURITY_INDEX_NAME, SINGLE_MAPPING_NAME, documentId)
                                 .setOpType(OpType.CREATE)
                                 .setSource(builder)
                                 .setRefreshPolicy(RefreshPolicy.WAIT_UNTIL)
                                 .request();
-                securityIndex.prepareIndexIfNeededThenExecute(ex -> listener.onFailure(traceLog("prepare security index", documentId, ex)),
-                    () -> executeAsyncWithOrigin(client, SECURITY_ORIGIN, IndexAction.INSTANCE, request,
-                        ActionListener.wrap(indexResponse -> listener.onResponse(new Tuple<>(userToken, refreshToken)),
-                            listener::onFailure))
-                );
+            } catch (IOException e) {
+                // unexpected exception
+                listener.onFailure(e);
+                return;
             }
-        }
-    }
-
-    /**
-     * Reconstructs the {@link UserToken} from the existing {@code userTokenSource} and call the listener with the {@link UserToken} and the
-     * refresh token string
-     */
-    private void reIssueTokens(Map<String, Object> userTokenSource,
-                               String refreshToken, ActionListener<Tuple<UserToken, String>> listener) {
-        final String authString = (String) userTokenSource.get("authentication");
-        final Integer version = (Integer) userTokenSource.get("version");
-        final Map<String, Object> metadata = (Map<String, Object>) userTokenSource.get("metadata");
-        final String id = (String) userTokenSource.get("id");
-        final Long expiration = (Long) userTokenSource.get("expiration_time");
-
-        Version authVersion = Version.fromId(version);
-        try (StreamInput in = StreamInput.wrap(Base64.getDecoder().decode(authString))) {
-            in.setVersion(authVersion);
-            Authentication authentication = new Authentication(in);
-            UserToken userToken = new UserToken(id, authVersion, authentication, Instant.ofEpochMilli(expiration), metadata);
-            listener.onResponse(new Tuple<>(userToken, refreshToken));
-        } catch (IOException e) {
-            logger.error("Unable to decode existing user token", e);
-            listener.onFailure(invalidGrantException("could not refresh the requested token"));
+            securityIndex.prepareIndexIfNeededThenExecute(ex -> listener.onFailure(traceLog("prepare security index", documentId, ex)),
+                    () -> executeAsyncWithOrigin(client, SECURITY_ORIGIN, IndexAction.INSTANCE, request, ActionListener
+                            .wrap(indexResponse -> listener.onResponse(new Tuple<>(userToken, refreshToken)), listener::onFailure)));
```

Comment:
I wonder if we should be sanity checking the index response for anything. I vaguely remember this being fine but it has been a while so it doesn't hurt to validate that again

Reply:
I added:
```
if (indexResponse.getResult() == Result.CREATED) {
  listener.onResponse(new Tuple<>(userToken, refreshToken));
} else {
  listener.onFailure(traceLog("create token", new ElasticsearchException("failed to create token document [{}]", indexResponse)));
}
```

Reply:
Had to push https://github.com/elastic/elasticsearch/pull/39808/commits/011033e2be90db4a137f7981e07cbd6a99c31eab to update the mocks to account for this.